src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,647 @@
+                .Select(parameterAndArguments => CreateLocalDeclarationStatement(syntaxGenerator, renameTable, parameterAndArguments));
+
+            var declarationsForVariableDeclarationArgumentQuery = parametersWithVariableDeclarationArgument
+                .Select(parameterAndName =>
+                    (TStatementSyntax)syntaxGenerator.LocalDeclarationStatement(
+                        parameterAndName.parameterSymbol.Type,
+                        parameterAndName.identifierName));
+
+            return declarationsQuery.Concat(declarationsForVariableDeclarationArgumentQuery).ToImmutableArray();
+        }
+
+        private bool ContainsAwaitExpression(TExpressionSyntax inlineExpression, TMethodDeclarationSyntax calleeMethodDeclarationNode)
+        {
+            // Check if there is await expression. It is used later if the caller should be changed to async
+            var awaitExpressions = inlineExpression
+                .DescendantNodesAndSelf()
+                .Where(node => node != null && _syntaxFacts.IsAwaitExpression(node))
+                .ToImmutableArray();
+            foreach (var awaitExpression in awaitExpressions)
+            {
+                var enclosingMethodLikeNode = GetEnclosingMethodLikeNode(awaitExpression);
+                if (calleeMethodDeclarationNode.Equals(enclosingMethodLikeNode))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static TStatementSyntax CreateLocalDeclarationStatement(
+            SyntaxGenerator syntaxGenerator,
+            ImmutableDictionary<ISymbol, string> renameTable,
+            (IParameterSymbol parameterSymbol, TExpressionSyntax expression) parameterAndExpression)
+        {
+            var (parameterSymbol, expression) = parameterAndExpression;
+            var name = renameTable.TryGetValue(parameterSymbol, out var newName) ? newName : parameterSymbol.Name;
+            return (TStatementSyntax)syntaxGenerator.LocalDeclarationStatement(parameterSymbol.Type, name, expression);
+        }
+
+        private static async Task<TExpressionSyntax> ReplaceAllSyntaxNodesForSymbolAsync(
+            Document document,
+            TExpressionSyntax inlineExpression,
+            SyntaxGenerator syntaxGenerator,
+            ImmutableDictionary<ISymbol, SyntaxNode> replacementTable,
+            CancellationToken cancellationToken)
+        {
+            var editor = new SyntaxEditor(inlineExpression, syntaxGenerator);
+
+            foreach (var (symbol, syntaxNode) in replacementTable)
+            {
+                var allReferences = await SymbolFinder
+                    .FindReferencesAsync(symbol, document.Project.Solution, cancellationToken)
+                    .ConfigureAwait(false);
+                var allSyntaxNodesToReplace = allReferences
+                    .SelectMany(reference => reference.Locations
```

Comment:
you should probably filter out implicit-references.  those aren't safe to generally get nodes for.

Reply:
Good catch